File: Array/2. Array insert at index - GFG/2.-array-insert-at-index.py
#User function Template for python3

class Solution:
    '''You need to insert the given element at the given index. 
    After inserting the elements at index, elements
    from index onward should be shifted one position ahead
     You may assume that the array already has sizeOfArray - 1
    elements.'''
    
    def insertAtIndex(self, arr, sizeOfArray, index, element):
        #Your code here
        # shifting all elements till index to the right by 1
        for i in range(sizeOfArray-1,index,-1):
            arr[i] = arr[i-1]
        
        # inserting element at the given index
        arr[index] = element
 
#{ 
#  Driver Code Starts
#Initial Template for Python 3


def main():
    T=int(input())
    while(T>0):
        sizeOfArray=int(input())
        arr=[int(x) for x in input().strip().split()]
        arr.append(-1)
        index,element=map(int,input().strip().split())
        
        Solution().insertAtIndex(arr,sizeOfArray,index,element)
        
        for i in range(sizeOfArray):
            print(arr[i],end=" ")
        print()
        T-=1

if __name__=="__main__":
    main()
# } Driver Code Ends

File: Array/2. Array insert at index - GFG/2.-array-insert-at-index.c
// { Driver Code Starts
//Initial Template for C

#include <stdio.h>


 // } Driver Code Ends
//User function Template for C

    // You need to insert the given element at the given index. 
    // After inserting the elements at index, elements
    // from index onward should be shifted one position ahead
    // You may assume that the array already has sizeOfArray - 1
    // elements.
    void insertAtIndex(int arr[], int sizeOfArray, int index, int element)
    {
        //Your code here
        
        // shifting all the elements to the right 
        // can be done only from right end of the array
        
        for( int i = sizeOfArray - 1; i > index; i--){
            arr[i] = arr[i-1];
        }
        arr[index] = element;
        
        
    }

// { Driver Code Starts.

int main() {
    int t;
    scanf("%d", &t);//Input testcases
    while(t--)
    {
        
        int sizeOfArray; 
        
        //input size of Array
        scanf("%d", &sizeOfArray);
        
        int arr[sizeOfArray];
        
        //input elements in the Array
        for(int i = 0; i < sizeOfArray-1; i++)
            scanf("%d", &arr[i]);
        
        int index;
        //input index
        scanf("%d", &index);
        int element;
        //input element
        scanf("%d", &element);
        //calling insertAtIndex() function
        insertAtIndex(arr, sizeOfArray, index, element); //Complete this function
        
        for(int i = 0; i < sizeOfArray; i++)//Printing the array for verification
            printf("%d ", arr[i]);
        
        printf("\n");
        
    }
    
	return 0;
}  // } Driver Code Ends